cpp/memilio/io/README.md

```diff
@@ -1,92 +1,95 @@
-     iterated over. Otherwise returns an error.
-- `obj.expect_optional("Name", Tag<T>{})`:
-     Returns boost::optional<T> if an optional value of type T can be found under the key "Name". The optional may contain a 
-     value or it may be empty. Otherwise returns an error. Note that for some formats a wrong key is indistinguishable from 
-     an empty optional, so make sure to provide the correct key.
- 
-Error handling:
----------------
-Errors are handled by returning error codes. The type IOStatus contains an error code and an optional string with additional 
-information. The type IOResult contains either a value or an IOStatus that describes an error. Operations that can fail return 
-an IOResult<T> where T is the type of the value that is produced by the operation if it is succesful. Except where necessary
+
+     - `obj.add_element("Name", t)`:
+          Stores an object `t` in the IOObject under the key "Name". If `t` is of basic type (i.e., int, string), IOObject
+          is expected to handle it directly. Otherwise, the object uses `mio::serialize` to get the data for `t`.
+     - `obj.add_list("Name", b, e)`:
+          Stores the elements in the range represented by iterators `b` and `e` under the key "Name". The individual elements are not named.
+          The elements are either handled directly by the IOObject or using `mio::serialize` just like `add_element`.
+     - `obj.add_optional("Name", p)`:
+          Stores the element pointed to by pointer `p` under the key "Name". The pointer may be null. Otherwise identical to add_element.
+     - `obj.expect_element("Name", Tag<T>{})`:
+          If an object of type T can be found under the key "Name" and can be deserialized, returns the object. Otherwise returns
+          an error. Analogously to serialization, the IOObject is expected to handle basic types directly and use `mio::deserialize`
+          otherwise.
+     - `obj.expect_list("Name", Tag<T>{})`:
+          If a list of objects of type T can be found under the key "Name" and can be deserialized, returns a range that can be
+          iterated over. Otherwise returns an error.
+     - `obj.expect_optional("Name", Tag<T>{})`:
+          Returns ```boost::optional<T>``` if an optional value of type T can be found under the key "Name". The optional may contain a
+          value or it may be empty. Otherwise returns an error. Note that for some formats a wrong key is indistinguishable from
+          an empty optional, so make sure to provide the correct key.
+
+## Error handling
+
+Errors are handled by returning error codes. The type IOStatus contains an error code and an optional string with additional
+information. The type IOResult contains either a value or an IOStatus that describes an error. Operations that can fail return
+an ```IOResult<T>``` where T is the type of the value that is produced by the operation if it is succesful. Except where necessary
```

Comment:
Does using three backticks make a difference to using one when the expression is used inline? Otherwise, I would use single backticks here instead.

Reply:
No doesn't .-)